modyn/supervisor/internal/triggers/datadrifttrigger.py

```diff
@@ -200,16 +238,32 @@ def _run_detection(self) -> tuple[bool, dict[str, MetricResult]]:
 
         # Compute embeddings
         assert self.embedding_encoder is not None
+
+        # tbd. reuse the embeddings as long as the reference window is not updated
         reference_embeddings = get_embeddings(self.embedding_encoder, reference_dataloader)
         current_embeddings = get_embeddings(self.embedding_encoder, current_dataloader)
         reference_embeddings_df = convert_tensor_to_df(reference_embeddings, "col_")
         current_embeddings_df = convert_tensor_to_df(current_embeddings, "col_")
 
         drift_results = {
-            **self.evidently_detector.detect_drift(reference_embeddings_df, current_embeddings_df),
-            **self.alibi_detector.detect_drift(reference_embeddings, current_embeddings),
+            **self.evidently_detector.detect_drift(reference_embeddings_df, current_embeddings_df, is_warmup),
+            **self.alibi_detector.detect_drift(reference_embeddings, current_embeddings, is_warmup),
         }
+
+        # make the final decisions with the decision engines
+        for metric_name, metric_result in drift_results.items():
+            distance = (
+                metric_result.distance
+                if isinstance(metric_result.distance, float)
+                else float(mean(metric_result.distance))
+            )
+            drift_results[metric_name].is_drift = self.decision_engines[metric_name].evaluate_decision(
+                distance, metric_result.is_drift
+            )
+
         logger.info(f"[DataDriftDetector][Dataset {self.dataloader_info.dataset_id}]" + f"[Result] {drift_results}")
+        if is_warmup:
+            return False, {}  # unused dummy
```

Comment:
not sure what you mean with this comment

Reply:
the dictionary is just a placeholder as we don't have any useful object to return in the warmup region, i remove the comment as it doesn't really help